scrappers/tbib/lib/Tbib.rb

```diff
@@ -0,0 +1,76 @@
+class Tbib
+	include RethinkDB::Shortcuts
+
+	def initialize()
+		@conn = r.connect(:host => $config.server_ip, :port => $config.server_port)
+		$log.log 'connected to database!'
+		@conn.use($config.database)
+		@mutex = Mutex.new
```

Comment:
This should probably be a semaphore (look into the `concurrent-ruby` gem) with a configureable limit (something like `pages-thread-limit` in your config.toml's `[other]`)

Reply:
semaphore not needed. threads are limited here:
```
while !procs.empty?
  $log.progress "Downloading images...", 1.0 - procs.length.to_f/ol
  b = procs.shift(8)
  b.each_with_index { |d, i| b[i] = d.call }
  b.each { |d| d.join }
  sleep 1
end
```

it takes 8 threads, starts them, and waits for them to finish, then starts the next 8.

Reply:
You should probably put a comment there explaining this behaviour